File: boat-common/src/test/java/test/java/xyz/srclab/common/base/FormatTest.java
package test.java.xyz.srclab.common.base;

import org.testng.Assert;
import org.testng.annotations.Test;
import xyz.srclab.common.base.Format;

/**
 * @author sunqian
 */
public class FormatTest {

    @Test
    public void testFormat() {
        String fastFormat = Format.fastFormat("This is {} {}.", "fast", "format");
        Assert.assertEquals(fastFormat, "This is fast format.");
        String printfFormat = Format.printfFormat("This is %s %s.", "printf", "format");
        Assert.assertEquals(printfFormat, "This is printf format.");
        String messageFormat = Format.messageFormat("This is {0} {1}.", "message", "format");
        Assert.assertEquals(messageFormat, "This is message format.");
    }
}

File: easy-projects/easy-common/src/test/kotlin/test/xyz/srclab/common/string/FormatTest.kt
package test.xyz.srclab.common.string

import org.testng.annotations.DataProvider
import org.testng.annotations.Test
import xyz.srclab.common.lang.format.FormatHelper
import xyz.srclab.test.doAssertEquals

object FormatTest {

    @Test(dataProvider = "fastFormatDataProvider")
    fun testFastFormat(actual: String, expected: String) {
        doAssertEquals(actual, expected)
    }

    @DataProvider
    fun fastFormatDataProvider(): Array<Array<*>> {
        return arrayOf(
            arrayOf(
                FormatHelper.fastFormat("123{}56{{}89{}"), "123{}56{{}89{}"
            ),
            arrayOf(
                FormatHelper.fastFormat("123{}56{{}89{}", 4, 7), "123456{789{}"
            ),
            arrayOf(
                FormatHelper.fastFormat("123{}56{{}89{}", 4, 7, NullPointerException()),
                "123456{789${NullPointerException()}"
            )
        )
    }

    @Test(dataProvider = "printfFormatDataProvider")
    fun testPrintfFormat(actual: String, expected: String) {
        doAssertEquals(actual, expected)
    }

    @DataProvider
    fun printfFormatDataProvider(): Array<Array<*>> {
        return arrayOf(
            arrayOf(FormatHelper.printfFormat("123%s56%%%s89", 4, 7), "123456%789")
        )
    }

    @Test(dataProvider = "messageFormatDataProvider")
    fun testMessageFormat(actual: String, expected: String) {
        doAssertEquals(actual, expected)
    }

    @DataProvider
    fun messageFormatDataProvider(): Array<Array<*>> {
        return arrayOf(
            arrayOf(FormatHelper.messageFormat("123{1}56{0}89", 7, 4), "123456789")
        )
    }
}